### ISSUE TITLE
[4.4-beta1] File picker differences

### ISSUE THREAD

**Issue by ausi** [contao/core-bundle#816]
While testing the new file picker I noticed some backwards incompatible differences between the current and the old version:

- With `eval.files = false` only the directories were shown in the picker. Now everything is shown and can be selected but only for the directories a radio button is visible.
- With `eval.filesOnly = true` directories are still selectable, even though no radio button is shown for them.
- The `eval.extensions` setting has no effect anymore.
- The `eval.path` setting has no effect anymore.

**Comment by leofeyer** [comment]
Of course, because now we do not have a separate picker anymore but the regular DC views with an additional checkbox or radio button. Those options are only supported in the old picker widget.

**Comment by ausi** [comment]
And we do not want to add support for them? IMO `eval.extensions` and `eval.path` are very important features of the `fileTree` widget.

**Comment by leofeyer** [comment]
They were features of the `FileSelector` class, not of the `FileTree` class. It is important to understand the difference. 

The file tree widget still works the same as in Contao 4.3, except it now uses the DCA picker instead of the file selector (which still exists in Contao 4.4 BTW). You want the `eval.extensions` and `eval.path` restrictions to apply to the DCA picker, too, however, as I said, the DCA picker is not a separate class but the regular DC views. It is like browsing the back end in the modal window.

I guess we could try to support `eval.extensions` and `eval.path` by modifying `$this->root`, but I don't like the idea to apply restrictions based on a GET parameter.

**Comment by Toflar** [comment]
> I guess we could try to support eval.extensions and eval.path by modifying $this->root, but I don't like the idea to apply restrictions based on a GET parameter.

We already init the picker itself based on GET params. Why shouldn't we do more? Imho this needs to go to `DC_Folder` like this (parent `initPicker()` must return `false` on early return to get this working):

```php
protected function initPicker()
{
    if (false === parent::initPicker()) {
        return false;
    }

    // Apply root based on eval of $GLOBALS['TL_DCA'][$this->strPickerTable]['fields'][$this->strPickerField]
}
```

And I do consider this to be very important too 👍

**Comment by leofeyer** [comment]
I have fixed the selection error in 02f206f1b6cd5521ced88dc48aa752e111fa854e.

**Comment by leofeyer** [comment]
I have begun to create a PR addressing the issues in #819.
